File: screen-switcher-sample/src/main/java/com/jaynewstrom/screenswitchersample/ApplicationComponent.java
package com.jaynewstrom.screenswitchersample;

import com.jaynewstrom.screenswitchersample.base.LeakWatcher;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(
        modules = {
                ApplicationModule.class
        }
)
public interface ApplicationComponent {
    LeakWatcher getLeakWatcher();
}

File: screen-switcher-sample/src/main/java/com/jaynewstrom/screenswitchersample/application/ApplicationComponent.kt
package com.jaynewstrom.screenswitchersample.application

import com.jaynewstrom.screenswitchersample.core.LeakWatcher
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationModule::class])
internal interface ApplicationComponent {
    val leakWatcher: LeakWatcher
}
